### ISSUE TITLE
Error in Error callbackId: HotSpotPlugin708704024 : TypeError: Cannot read property '1' of null

### ISSUE THREAD

**Issue by mcintosh101** [hypery2k/cordova-hotspot-plugin#43]
Hi,

I can't seem to figure out how this is happening, I am using the plugin as suggested but I get back a lot of these errors as I try to poll a list of IP Addresses to see if they are available.

Here is some example code I am using:

``` javascript
// Get the tablet net configuration
cordova.plugins.hotspot.getNetConfig((config) => {
    let gateway = config.deviceIPAddress.split('.');

    // Prepare the IP Gateway
    const assignedIPID = gateway.pop();
    gateway = gateway.join('.');

    // Poll all available IPs
    for (let i = 0; i < 256; i++) {
        const ip = `192.168.15.${i}`;

        if (i !== assignedIPID) {
            // If the IP doesn't match the tablet, ping it for a response
            cordova.plugins.hotspot.pingHost(ip, () => {
                // Response received, get the mac address of the device
                cordova.plugins.hotspot.getMacAddressOfHost(ip, (macAddress) => {
                    // Got the Mac Address, check to see if it is a Raspberry Pi
                    if (macAddress.substring(0, 8) === 'b8:27:eb') {
                        pieTin.push(ip);
                    }
                });
            });
        }
    }
});
```

![image](https://cloud.githubusercontent.com/assets/2189617/17699314/ddfdb188-63b7-11e6-807d-a8122c0072c8.png)

**Ionic version**: 1.7.16
**Cordova version**: 6.2.0
**Cordova Plugin Hotspot version**: 1.2.5
**Android API version**: 23

**Comment by hypery2k** [comment]
can you post without the magnification turned on?

**Comment by mcintosh101** [comment]
I have magnification turned on?

Here is the output from launch going top to bottom on the console:

```
 Error in Error callbackId: HotSpotPlugin1921838211 : TypeError: Cannot read property '1' of null
cordova.js?ionicCachebuster=27571:314 Uncaught TypeError: Cannot read property '1' of null
cordova.js?ionicCachebuster=27571:312 Error in Error callbackId: HotSpotPlugin1921838223 : TypeError: Cannot read property '1' of null
```

![image](https://cloud.githubusercontent.com/assets/2189617/17702884/a5473048-63c7-11e6-88ef-d72355d53bb1.png)

Thanks for the response!

**Comment by hypery2k** [comment]
sorry i mean minification ;) Can you debug the app without minifying your app? If you can somehow share your project I can take a look at it

**Comment by mcintosh101** [comment]
Hah, cool, I see what you mean. The error is coming from the Cordova JS file which isn't being minified, I've just unminified everything and sent it to the tablet and the errors are exactly the same, no extra info is given.

**Comment by hypery2k** [comment]
But it's also uglified, so the names of variables and functions are changed. If you share your project I can take a look, feel fee to drop me an email, see my github profile for details

**Comment by mcintosh101** [comment]
Thanks - I'll email you some information.

**Comment by hypery2k** [comment]
It looks like it was when the pingHost function is called inside HostSpotPlugin.js, the parseResponse runs and responds with the parameter response.

To take it step-by-step through the function, here is an example of what response is when it normally fails:

 PING 192.168.15.3 (192.168.15.3) 56(84) bytes of data.From 192.168.15.191: icmp_seq=1 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=2 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=3 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=4 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=5 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=6 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=7 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=8 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=9 Destination Host Unreachable--- 192.168.15.3 ping statistics ---10 packets transmitted, 0 received, +9 errors, 100% packet loss, time 9024mspipe 4

The first thing that happens is it defined the stats variable which will equal:

5.3 (192.168.15.3) 56(84) bytes of data.From 192.168.15.191: icmp_seq=1 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=2 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=3 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=4 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=5 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=6 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=7 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=8 Destination Host UnreachableFrom 192.168.15.191: icmp_seq=9 Destination Host Unreachable--- 192.168.15.3 ping statistics ---10 packets transmitted, 0 received, +9 errors, 100% packet loss, time 9024mspipe 4

Next the function defines the result object, which comes back False.

**Comment by hypery2k** [comment]
I've fixed it in the latest version 1.2.6
